documentation/metadata-editing.md

```diff
@@ -0,0 +1,87 @@
+
+
+
+## How to edit the metadata of a resource
+
+Editing the metadata of a resource is performed by editing the auxiliary metadata resource directly. This can only be done using `PATCH` requests (see [example workflow](#workflow)).
+Thus it is forbidden to perform a `PUT` request on metadata. The reason of this approach is that some metadata is [protected by the server](https://solidproject.org/TR/protocol#resource-containment) and other metadata is generated by the server. A `PUT` request, according to the HTTP specification, replaces the state of the target resource entirely by the request payload. This is not possible with metadata resources and therefore not allowed.
+
+### Protected metadata
+
+There is some metadata that is forbidden to be modified. The CSS will throw an error when trying to change this protected metadata.
+
+Therefore any change PATCH request with a query that will result into modifying the **storage indication** or the **containment triples** will result into a 409 `ConflictHttpError`.
+
+#### Storage indication
+
+The root of pod is indicated with `pim:Storage`. Any attempt to change metadata regarding this triple will fail.
+
+Thus the following query on the root metadata resource will fail:
+
+```shell
+curl --location --request PATCH 'http://localhost:3000/.meta' \
+--header 'Content-Type: text/n3' \
+--data-raw '@prefix solid: <http://www.w3.org/ns/solid/terms#>.
+<> a solid:InsertDeletePatch;
+solid:deletes { <http://localhost:3000/> a <http://www.w3.org/ns/pim/space#Storage> }.'
+```
+
+#### Containment triples
+
+A container can contain several resources, indicated with the predicate `ldp:contains`. Any attempt of adding or removing a resource via editing metadata will fail.
+
+Thus following example query will fail:
+
+```shell
+curl --location --request PATCH 'http://localhost:3000/.meta' \
+--header 'Content-Type: text/n3' \
+--data-raw '@prefix solid: <http://www.w3.org/ns/solid/terms#>.
+<> a solid:InsertDeletePatch;
+solid:inserts { <http://localhost:3000/> <http://www.w3.org/ns/ldp#contains> <http://localhost:3000/resource> }.'
+```
+
+## Example of a workflow for editing a metadata resource {#workflow}
+
+To give some context, assume we have started the CSS with the default configuration and have created a resource which has as URL `http://localhost:3000/resource`.
+
+We don't know the location of the metadata resource, so first we send a `HEAD` request to the resource to retrieve the metadata resource URL.
+
+```shell
+curl --location --head 'http://localhost:3000/resource'
+```
+
+which will at least the following response
+
+```shell
+HTTP/1.1 200 OK
+Link: <http://localhost:3000/resource.meta>; rel="describedby"
+```
+
+Now that we have the URL of the metadata resource, we create a query of what we want to change in the metadata of the resource (e.g. inserting `<s> <p> <o>.` ). This can be either with a SPARQL Update
```

Comment:
You can just put a link to `example-requests.md` for people that want to know how to do a patch request. No need to explain again here how a patch works.